src/components/sandbox/AppSandbox.vue

```diff
@@ -632,6 +665,30 @@ module.exports = {
             }
             return true;
         },
+        handleInstallAppRequest: function(header, path, apiMethod, data, hasFormData, params) {
+            let that = this;
+            if (!this.isAppGalleryMode || !hasFormData || apiMethod != 'POST') {
+                console.log('Install App API call is invalid');
+                that.buildResponse(headerFunc(), null, that.ACTION_FAILED);
+                return;
+            }
+            let requestBody = JSON.parse(new TextDecoder().decode(data));
```

Comment:
This looks like an app can install any other app?

Reply:
functionality only available if isAppGalleryMode is set.
I have added a further restriction. Now using a normally invalid app name to trigger mode to guard against a malicious user.

Reply:
Can you explain the security assumptions and model here and how it's enforced regarding installing an app?

Reply:
Maybe "able to install apps" should itself be a permission?

Reply:
The Appsandbox is responsible for ensuring an app has api access via the permissions
The htmlviewer is an internal app that has no permissions and therefore no access to any api
The install-app api call is only relevant when the recommended apps button on the launcher page is pressed.
The recommended apps page is launched in the htmlviewer via the call to launchApp on line 467 of Launcher.vue
The App Sandbox recognises the app-gallery token and sets isAppGalleryMode=true.
It is not possible via construction for any other app to set isAppGalleryMode

Reply:
Does it rely on $ not being allowing in app name? If so, where do we enforce that?

Reply:
Yes. we restrict the characters allowed in an app name. we enforce it in mixins/sandbox/index.js validateDisplayName(). this is called from verifyJSONFile() which is in turn called by AppInstall.vue when installing an app.

Reply:
In the future I'd like to let anyone write their own "app store" which could request the "install apps" permission. Does any of this block that or make it harder? I'd like to keep that open.

Reply:
Yes, that path is available down the track.